Google.PowerShell/Common/GcloudCmdlet.cs

```diff
@@ -53,6 +53,34 @@ public static BaseClientService.Initializer GetBaseClientServiceInitializer()
             };
         }
 
+        protected Tuple<string, ProviderInfo> GetFullPath(string filePath)
+        {
+            // Try to resolve the path using PowerShell.
+            try
+            {
+                ProviderInfo provider = null;
+                string[] result = GetResolvedProviderPathFromPSPath(filePath, out provider).ToArray();
+
+                // Only return the resolved path if there are no ambiguities.
+                // If path contains wildcards, then it may resolved to more than 1 path.
+                if (result?.Length == 1)
+                {
+                    return new Tuple<string, ProviderInfo>(result[0], provider);
+                }
+            }
+            catch (ItemNotFoundException itemEx)
+            {
+                // This exception is thrown if the path does not exist.
+                // We can still resolve the path with GetUnresolvedProviderPathFromPSPath.
+                ProviderInfo provider = null;
+                PSDriveInfo psDrive = null;
+                string path = SessionState.Path.GetUnresolvedProviderPathFromPSPath(itemEx.ItemName, out provider, out psDrive);
```

Comment:
This is a little confusing.  The path does not exist but we can still find it?

Can you clarify what is happening here?